accepted/future-releases/nnbd/feature-specification.md

```diff
@@ -493,21 +497,20 @@ fields on `Object`.
 It is an error to call an expression whose type is potentially nullable and not
 `dynamic`.
 
-It is an error if a top level variable or static variable with a non-nullable type
-has no initializer expression unless the variable is marked with the `late` modifier.
-
-It is an error if a class declaration declares an instance variable with a potentially
-non-nullable type and no initializer expression, and the class has a generative constructor
-where the variable is not initialized via an initializing formal or an initializer list entry,
-unless the variable is marked with the `late` modifier.
+It is an error if a top level variable or static variable with a non-nullable
+type has no initializer expression unless the variable is marked with a
+`late` or `external` modifier.
 
-It is an error if a mixin declaration or a class declaration with no generative constructors
-declares an instance variable with a potentially non-nullable type and no initializer expression
-unless the variable is marked with the `late` modifier.
+It is an error if a class declaration declares an instance variable with a
+potentially non-nullable type and no initializer expression, and the class has
+a generative constructor where the variable is not initialized via an
+initializing formal or an initializer list entry, unless the variable is marked
+with a `late`, `abstract`, or `external` modifier.
 
-It is an error to derive a mixin from a class declaration which contains
-an instance variable with a potentially non-nullable type and no initializer expression
-unless the variable is marked with the `late` modifier.
+It is an error if a mixin declaration or a class declaration with no generative
+constructors declares an instance variable without an initializing expression
+which is final or whose type is potentially non-nullable, unless the variable
+is marked with a `late`, `abstract`, or `external` modifier.
```

Comment:
I guess this situation for a *class* is only an issue if the class can be used as a mixin (so only if it's superclass is `Object`, and only as long as we allow extracting mixins from class declarations.
A class with no generative constructor cannot be instantiated, so it can never have an uninitialized instance variable.

Reply:
> only an issue if the class can be used as a mixin

Agreed, and I was thinking about this distinction as well. I think there's little value, however, in allowing an uninitialized final instance variable in this very special case where it is guaranteed to never exist. Such a class should probably be abstract, and the final instance variable could then be expressed as an abstract getter (or an abstract final instance variable).

If you wish to pursue that we allow it then I'd handle that in a separate PR.